lib/db/leveldb_dbinstance.go

```diff
@@ -144,6 +153,31 @@ func (db *Instance) updateFiles(folder, device []byte, fs []protocol.FileInfo, m
 	}
 }
 
+func (db *Instance) addSequences(folder []byte, fs []protocol.FileInfo) {
+	t := db.newReadWriteTransaction()
+	defer t.close()
+
+	var sk []byte
+	var dk []byte
```

Comment:
Don't these need a make()? The code allocates a new slice if the original one is too small, but that doesn't affect the existing slice, just creates a new one, so I suspect this allocates on every iteration?

Reply:
Good catch. I only use the `...Into` functions as arguments instead of assigning. That's going to change due to https://github.com/syncthing/syncthing/pull/4906#discussion_r184605927 and then I think it should be fine.

Reply:
You can just assign the return value back to sk in the loop, and pass in sk to tge db op.

Reply:
Also, it's interesting if we don't have something like this already elsewhere.

Reply:
That's what I do since the review commit, see some lines below. And good instincts again, I do the same mistake in the migration code...

Reply:
And yes, there was quite a lot of potential for key reuse - I addressed it in https://github.com/syncthing/syncthing/pull/4906/commits/ab9dc3834a50ef5b675ac35aaa79ee5c3772614e